api/src/main/java/org/openmrs/api/impl/TagServiceImpl.java

```diff
@@ -0,0 +1,158 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api.impl;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.OpenmrsObject;
+import org.openmrs.api.UserService;
+import org.openmrs.Tag;
+import org.openmrs.api.TagService;
+import org.openmrs.api.db.TagDAO;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.ArrayList;
+
+@Transactional(readOnly = true)
+public class TagServiceImpl extends BaseOpenmrsService implements TagService {
+	
+	protected final Log log = LogFactory.getLog(this.getClass());
+	
+	private TagDAO dao;
+	
+	UserService userService;
+	
+	/**
+	 * Injected in moduleApplicationContext.xml
+	 */
+	public void setDao(TagDAO dao) {
+		this.dao = dao;
+	}
+	
+	/**
+	 * Injected in moduleApplicationContext.xml
+	 */
+	public void setUserService(UserService userService) {
+		this.userService = userService;
+	}
+	
+	@Override
+	@Transactional
+	public Tag getTagByUuid(String uuid) {
+		return dao.getTagByUuid(uuid);
+	}
+	
+	@Override
```

Comment:
This code doesn't work outside of unit tests, because you have readOnly set to true, you need to set it to false for all these methods that need to write to the DB